pubtools/_pulp/tasks/delete.py

```diff
@@ -0,0 +1,709 @@
+
+        # collect items
+        f = f_map(cleared_repos_f, self.record_clears)
+        self.to_await.extend(f.result())
+
+        # return affected repos
+        return cleared_repos_f
+
+    def delete_files(self, file_names, repos):
+        # get files from Pulp
+        files_f = self.get_files(file_names)
+
+        # map files to repos
+        f = f_map(files_f, partial(self.map_to_repo, repos=repos, unit_attr="path"))
+        repo_map_f = f_map(f, self.log_units)
+
+        # remove from repos
+        cleared_repos_f = f_map(repo_map_f, self.remove_files)
+
+        # collect items
+        f = f_map(cleared_repos_f, self.record_clears)
+        self.to_await.extend(f.result())
+
+        # return affected repos
+        return cleared_repos_f
+
+    def delete_rpms(self, rpm_names, signing_keys, repos):
+        # get rpms from Pulp
+        rpms_f = self.get_rpms(rpm_names, signing_keys)
+
+        # map rpms to repos
+        f = f_map(rpms_f, partial(self.map_to_repo, repos=repos, unit_attr="filename"))
+        repo_map_f = f_map(f, self.log_units)
+
+        # remove rpms from repos
+        cleared_repos_f = f_map(repo_map_f, self.remove_rpms)
+
+        # collect items
+        f = f_map(cleared_repos_f, self.record_clears)
+        self.to_await.extend(f.result())
+
+        # return affected repos
+        return cleared_repos_f
+
+    def log_units(self, maps):
+        missing = []
+        repo_map, unit_map = maps
+
+        for unit, runit_item in sorted(unit_map.items()):
+            repos = runit_item.repos
+            if not repos:
+                missing.append(unit)
+            LOG.info("Deleting %s from %s", unit, repos)
+
+        if missing:
+            LOG.warning(
+                "Unit(s) doesn't belong to any requested repos %s: %s",
+                repos,
+                ",".join(sorted(missing)),
+            )
+            # TODO: collect units "SKIPPING"
+
+        return repo_map
+
+    def log_items(self, repo_map):
+        item_map = {}
+
+        for repo, items in repo_map.items():
+            for item in items:
+                item_map.setdefault(item, []).append(repo)
+
+        for item, repos in sorted(item_map.items()):
+            LOG.info("Deleting %s from %s", item.name, repos)
+
+        return repo_map
+
+    @step("Process push items")
+    def process_pushitems(self, pushitems, repos):
+        repo_map = {}
+
+        for item in pushitems:
+            if not isinstance(item, RpmPushItem):
+                LOG.warning("%s is not and RpmPushItem. Skipped for removal", item.name)
+                continue
+            mapped = False
+            for repo in repos or item.dest:
+                if repo in item.dest:
+                    repo_map.setdefault(repo, []).append(item)
+                    mapped = True
+
+            if not mapped:
+                LOG.warning(
+                    "%s is not present in any repo: %s", item.name, ", ".join(repos)
```

Comment:
This warning isn't right: item.dest isn't telling you which repos the item is *present in*.  It's telling you which repos Errata Tool (or other item source) wants the item to be pushed to.  I think it's a symptom of the overall concept here being wrong, the command should be driven only by the arguments and the state of Pulp, not the state of ET.

Reply:
Yes, might not need this if taking the above suggested approach.

Reply:
Removed it as it wasn't required with the new workflow.